### PR TITLE
[Semantic API] Add testcases related to query expressions

### PR DESCRIPTION
**Pull request by DamithaSenevirathne**
Related to  #30437

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/queryexpressions/QueryExpressionTest.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
+/*
```
Comment (pubudu91) [inline tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/queryexpressions/QueryExpressionTest.java:1]: Shall we refactor the package name? `queryexpressions->expressions`

File: tests/ballerina-compiler-api-test/src/test/resources/test-src/query-expression/query_expression_tests.bal [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+function testQueryExpression() {
+    Student s1 = {name: "Foo", age: 1, gpa: 2.1};
+    Student s2 = {name: "Bar", age: 2, gpa: 3.2};
+    Student s3 = {name: "Baz", age: 3, gpa: 1.3};
+
+    Student[] students = [s1, s2, s3];
+
+    var x = from var st in students
+        where st.name == "Foo"
+        select {name: st.name};
```
Comment (pubudu91) [inline tests/ballerina-compiler-api-test/src/test/resources/test-src/query-expression/query_expression_tests.bal:32]: Shall we also try a few more variations of the query expressions as well? Covering all possible clauses in query expessions?

Comment (DamithaSenevirathne) [reply tests/ballerina-compiler-api-test/src/test/resources/test-src/query-expression/query_expression_tests.bal]: updated in 0aea9c28ba0c4281afbef57bc06d3ba1306f4d60

File: tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/queryexpressions/QueryExpressionTest.java (hunk 1 above)
Comment (DamithaSenevirathne) [reply tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/queryexpressions/QueryExpressionTest.java]: Updated in 0aea9c28ba0c4281afbef57bc06d3ba1306f4d60

File: tests/ballerina-compiler-api-test/src/test/resources/test-src/query-expression/query_expression_tests.bal [hunk 3]
```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (pubudu91) [inline tests/ballerina-compiler-api-test/src/test/resources/test-src/query-expression/query_expression_tests.bal:1]: Shall we organize these test files in the same directory structure as well? i.e., `test-src/expressions/query_expression_tests.bal`

Comment (DamithaSenevirathne) [reply tests/ballerina-compiler-api-test/src/test/resources/test-src/query-expression/query_expression_tests.bal]: Updated in 011f37a60e1ca34dccec4e3f2700ffe19ab92cd9

**Comment by pubudu91** [comment]
Can you rebase this with the master as well?

**Review by pubudu91 — APPROVED** [review]

**Pull request merged**